sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala

```diff
@@ -76,7 +76,7 @@ object ExplainUtils extends AdaptiveSparkPlanHelper {
    *   1. Generates the explain output for the input plan excluding the subquery plans.
    *   2. Generates the explain output for each subquery referenced in the plan.
    */
-  def processPlan[T <: QueryPlan[T]](plan: T, append: String => Unit): Unit = {
+  def processPlan[T <: QueryPlan[T]](plan: T, append: String => Unit): Unit = synchronized {
```

Comment:
We should add more comments to explain it. Ideally this is a no-op as different explain actions operate on different plan instances, but cached plan is an exception.

Reply:
added comment

Reply:
`children` is `def` too, but I don't see we create local variables...